geoengine/workflow.py

```diff
@@ -566,10 +566,19 @@ def create_xarray(record_batch: pa.RecordBatch) -> xr.DataArray:
 
             time = TimeInterval.from_response(json.loads(metadata[b'time']))
 
+            numpy_data = arrow_array.to_numpy(
+                zero_copy_only=False,  # cannot zero-copy as soon as we have nodata values
+            )
+            numpy_mask = arrow_array.is_null(
+                nan_is_null=False  # nan is not no data
+            ).to_numpy(
+                zero_copy_only=False  # cannot zero-copy with bools
+            )
+
+            numpy_masked_data = np.ma.masked_array(numpy_data, mask=numpy_mask).reshape(x_size, y_size)
```

Comment:
Thanks for finding this out.

Could you add a test that verifies that I don't change it back for optimization reasons in the future?

Reply:
well it was a bit more complex but now you can choose how to use the data. For Xarray the data is still converted to float and nan is set as "nodata". For numpy we can now produce masked arrays with the correct type.